Bluetooth: host: Add API to run conn CTE response HCI command

ppryga-nordic opened the pull request:
Add host API to allow execution of HCI_LE_Connection_CTE_Response_-
Enable HCI command.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga commented:
@jhedberg @alwa-nordic could you take a look?

jhedberg commented on subsys/bluetooth/host/direction.c:
```diff
@@ -767,3 +813,40 @@ int bt_df_conn_cte_rx_disable(struct bt_conn *conn)
+#if defined(CONFIG_BT_DF_CONNECTION_CTE_RSP)
+static int bt_df_set_conn_cte_rsp_enable(struct bt_conn *conn, bool enable)
+{
+	CHECKIF(!conn)
+	{
```
minor: the { should be at the end of the first line (basically same coding style as for an if-statement)

ppryga replied:
Thanks, I've corrected it here and in couple other functions also.

jhedberg replied:
Thanks, however please keep any coding style fixes to existing code in separate commits (they can still be part of the same PR though)

ppryga replied:
Yeah, will do :-)

jhedberg reviewed (changes requested):
Looks pretty good. Just one minor coding style comment.

ppryga-nordic committed:
Bluetooth: host: Add API to run conn CTE response HCI command

Add host API to allow execution of HCI_LE_Connection_CTE_Response_-
Enable HCI command.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic committed:
Bluetooth: host: df: Correct style for CHECKIF statements

Correct wrong coding style used in couple of plces where
CHECKIF marco is used.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga commented:
Fixed wrong parameter in `bt_hci_cmd_create` call. Was `BT_HCI_OP_LE_CONN_CTE_REQ_ENABLE`, should be `BT_HCI_OP_LE_CONN_CTE_RSP_ENABLE`.
